packages/aws-cdk-lib/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -143,28 +143,138 @@ class FileAccessLog extends AccessLog {
+  constructor(filePath: string, loggingFormat?: LoggingFormat) {
     super();
     this.filePath = filePath;
+    // For now we have the same setting for Virtual Gateway and Virtual Nodes
+    this.virtualGatewayLoggingFormat = loggingFormat?.bind().formatConfig;
+    this.virtualNodeLoggingFormat = loggingFormat?.bind().formatConfig;
   }
 
   public bind(_scope: Construct): AccessLogConfig {
     return {
       virtualNodeAccessLog: {
         file: {
           path: this.filePath,
+          format: this.virtualNodeLoggingFormat,
         },
       },
       virtualGatewayAccessLog: {
         file: {
           path: this.filePath,
+          format: this.virtualGatewayLoggingFormat,
         },
       },
     };
   }
 }
 
+/**
+ * All Properties for Envoy Access Logging Format for mesh endpoints
+ */
+export interface LoggingFormatConfig {
+  /**
+   * CFN configuration for Access Logging Format
+   *
+   * @default - no access logging format
+   */
+  readonly formatConfig?: CfnVirtualNode.LoggingFormatProperty;
+}
+
+/**
+ * Configuration for Envoy Access Logging Format for mesh endpoints
+ */
+export abstract class LoggingFormat {
+  /**
+   * Generate logging format from text pattern
+   */
+  public static fromText(text: string): LoggingFormat {
+    return new TextLoggingFormat(text);
+  }
+  /**
+   * Generate logging format from json key pairs
+   */
+  public static fromJson(jsonLoggingFormat :{[key:string]: string}): LoggingFormat {
+    const json: JsonFormatRef[] = [];
+    if (Object.keys(jsonLoggingFormat).length == 0) {
+      throw new Error('Json key pairs cannot be empty.');
+    }
+
+    for (const key in jsonLoggingFormat) {
+      json.push(new JsonFormatRef(key, jsonLoggingFormat[key]));
+    };
+
+    return new JsonLoggingFormat(json);
+  };
+
+  /**
+   * Called when the Access Log Format is initialized. Can be used to enforce
+   * mutual exclusivity with future properties
+   */
+  public abstract bind(): LoggingFormatConfig;
+}
+
+/**
+ * Configuration for Json logging format
+ */
+class JsonLoggingFormat extends LoggingFormat {
+  /**
+  * Json pattern for the output logs
+  *
+  * @default - no json format specified
+  */
+  private readonly json?: Array<CfnVirtualNode.JsonFormatRefProperty>;
+  constructor(json: Array<CfnVirtualNode.JsonFormatRefProperty>) {
+    super();
+    this.json = json;
+  }
+
+  bind(): LoggingFormatConfig {
+    return {
+      formatConfig: {
+        json: this.json,
+      },
+    };
+  }
+}
+
+class TextLoggingFormat extends LoggingFormat {
+  /**
+  * Json pattern for the output logs
+  *
+  * @default - no text format specified
+  */
+  private readonly text?: string;
+  constructor(text: string) {
+    super();
+    this.text = text;
+  }
+
+  public bind(): LoggingFormatConfig {
+    return {
+      formatConfig: {
+        text: this.text,
+      },
+    };
+  }
+}
+
+/**
+  * Key Value pair json reference for json Format
+  */
+class JsonFormatRef {
```

Comment:
Why do you need to create this new type?

Reply:
Deleted that, that's part of our api design for other parts